### DISCUSSION TITLE
[Animation] Skeletal mesh parent bone index assertion

### DISCUSSION THREAD

**Discussion by bleutn** [mholtkamp/octave#12] [Q&A]
Hello,

I'm trying to import a skeletal mesh but i got an assertion about parent bone index

[Assert] parentBoneIndex < (int32_t)i, SkeletalMesh.cpp, line 714

then editor crashes

any ideas ?

Bl3ut3n

**Comment by mholtkamp** [comment]
Hmm, in the code it looks like I'm assuming a bone's parent bone should be encountered first when iterating over all the bones, but it looks like that assumption isn't always true. Would you be able to provide the .glb file you are trying to import? No worries if you can't (paid/proprietary asset), but it will make testing a little easier. I'll try to look at this bug tonight, thanks for pointing it out!

**Reply by mholtkamp** [reply]
I just pushed a change that might fix the issue.
https://github.com/mholtkamp/octave/commit/099d37f9b3ea883b43784a36c6fa69fcf8c35d17

**Reply by bleutn** [reply]
I'm not using blender but 3dsmax. I saw in your engine that u got .DAE format models so i tried to export the model with this file format. However, u learned me that i can use glb file so i tried and i had no error. But there is something a bit annoying when importing models is that only 1 material is counted for loading geometry. Is it possible for u to allow multiples materials on geometries to be loaded by model ? By the way, models i'm using are free on mixamo. I used DAE format export but there was a bug (file weighted 6MB i dunno why) with this export method on mixamo so i exported firstly fbx then reexported as dae using max
